anyutil/any.go

```diff
@@ -61,3 +67,40 @@ func MarshalFrom(dst *anypb.Any, src proto.Message, opts proto.MarshalOptions) e
 	dst.Value = b
 	return nil
 }
+
+// Unpack unpacks the message inside an any. It tries first to unpack using
+// anypb.UnmarshalNew (which uses protoregistry.GlobalTypes under the hood).
+// If the message is not resolved in the proto v2 global registry, it tries to
+// unpack using dynamicpb and the given protoFiles.
+func Unpack(any *anypb.Any, protoFiles *protoregistry.Files) (proto.Message, error) {
```

Comment:
I think we should add a parameter with the `Types` instance. This code uses `GlobalTypes` implicitly.

Also, let's use `protodesc.Resolver` and `protoregistry. MessageTypeResolver`